ruby/ql/src/queries/security/cwe-094/CodeInjection.ql

```diff
@@ -21,6 +21,13 @@ import DataFlow::PathGraph
 from Configuration config, DataFlow::PathNode source, DataFlow::PathNode sink, Source sourceNode
 where
   config.hasFlowPath(source, sink) and
-  sourceNode = source.getNode()
+  sourceNode = source.getNode() and
+  // removing duplications of the same path, but different flow-labels.
+  sink =
+    min(DataFlow::PathNode otherSink |
+      config.hasFlowPath(any(DataFlow::PathNode s | s.getNode() = source.getNode()), otherSink)
+    |
+      otherSink order by otherSink.getState()
+    )
```

Comment:
Seems like this should be fixed in the shared libraries, the same way we did in JS. I've opened an internal issue for it.